### ISSUE TITLE
Twisted should have a CONTRIBUTING document in it's root directory.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6715]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)| @dreid reported|
|-|-|
|Trac ID|trac#6715|
|Type|enhancement|
|Created|2013-08-29 23:13:37Z|
|Branch|https://github.com/twisted/twisted/tree/contributing-6715|

This should include the contribution guidelines (a link to the existing contribution guidelines is acceptable.)

It should also include a warning that we are not currently capable of dealing with pull requests submitted via Github.

Github will also call out the contributing document when users actually attempt to submit pull requests.  https://github.com/blog/1184-contributing-guidelines

The root of the repo is generally a good place for this information as it is often the first interaction point for new contributors.

<details><summary>Searchable metadata</summary>

```
trac-id__6715 6715
type__enhancement enhancement
reporter__dreid dreid
priority__normal normal
milestone__ 
branch__branches_contributing_6715 branches/contributing-6715
branch_author__dreid dreid
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1377818017000000 1377818017000000
changetime__1377833116000000 1377833116000000
version__None None
owner__dreid dreid

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_1"></a>|@dreid commented|
|-|-|

(In [39812]) Branching to contributing-6715.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jml|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_4"></a>|@jml set owner to @dreid|
|-|-|

Thanks for doing this. It's a good idea.

 1. Please add a NEWS file in topfiles/
 2. "Patches can be submitted as pull requests, but if you don't file a ticket, it's unlikely that we'll notice your contribution." -- I'm not sure what everything before the 'but' is supposed to mean. Perhaps, "Your ticket may link to a pull request, but if you don't ..." is clearer?

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_6"></a>|@dreid commented|
|-|-|

Thanks for the review jml.

I added a .misc file, and removed the section about submitting pull requests if they had an associated ticket.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jml|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_8"></a>|@jml set owner to @dreid|
|-|-|

LGTM. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_9"></a>|@dreid set status to `closed`|
|-|-|

(In [39820]) Merge contributing-6715: Add a CONTRIBUTING document to the repository root.

Author: dreid
Reviewer: jml
Fixes: [#6715](https://github.com/twisted/twisted/issues/6715)
